azurerm/resource_arm_application_gateway.go

```diff
@@ -176,6 +176,26 @@ func resourceArmApplicationGateway() *schema.Resource {
 							},
 						},
 
+						"connection_draining": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:     schema.TypeBool,
+										Required: true,
+									},
+
+									"drain_timeout_sec": {
+										Type:         schema.TypeInt,
+										Required:     true,
```

Comment:
@katbyte i had left this as `Optional: true` since there is an `enabled` option. 

I thought that if user would be able to set `enabled = false` they would not need to specify `drain_timeout_sec `

Reply:
The API does require it however. If the user just wishes to have it disabled leaving the block out should be good enough 🙂